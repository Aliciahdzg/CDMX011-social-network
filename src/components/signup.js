import { onNavigate } from '../app.js';
import { ButtonGoogle } from './googlebutton.js';
import { spacebetween } from './spacebetween.js';
import { logo } from './logo.js';
import { signUp } from '../firebase.js';

export const Signup = () => {
  // se crean todos los elementos de DOM
  const signupView = document.createElement('section');
  const sectionChild = document.createElement('section');
  const signupForm = document.createElement('form');
  const inputMail = document.createElement('input');
  const inputPassword = document.createElement('input');
  const displayPassword = document.createElement('div');
  const togglePassword = document.createElement('img');
  const buttonSignup = document.createElement('button');
  const loginButton = document.createElement('button');
  const spaceButtons = document.createElement('br');
  // Section signup que contiene todos los elementos de la vista general
  signupView.setAttribute('class', 'grid-father');
  // Section que contiene logo y form
  sectionChild.setAttribute('class', 'container');
  signupView.appendChild(logo());
  signupView.appendChild(sectionChild);
  signupForm.id = 'signup';
  sectionChild.appendChild(signupForm);
  // Input del email
  inputMail.setAttribute('type', 'email');
  inputMail.setAttribute('class', 'form-elements');
  inputMail.setAttribute('placeholder', 'Correo');
  inputMail.setAttribute('name', 'email');
  inputMail.id = 'emailUser';
  signupForm.appendChild(inputMail);
  // A continuacion el input de password
  inputPassword.setAttribute('type', 'password');
  inputPassword.setAttribute('placeholder', 'Contraseña');
  inputPassword.className = 'input-password';
  inputPassword.setAttribute('name', 'password-user');
  inputPassword.id = 'passwordUser';
  inputPassword.autocomplete = 'on';
  displayPassword.setAttribute('class', 'form-elements');
  displayPassword.id = 'displayPassword';
  togglePassword.id = 'togglePassword';
  togglePassword.className = 'eye-img';
  togglePassword.src = 'images/eye-regular.jpeg';
  togglePassword.type = 'button';
  displayPassword.appendChild(inputPassword);
  displayPassword.appendChild(togglePassword);
  signupForm.appendChild(displayPassword);
  // listener de ocultar y mostrar contraseña
  togglePassword.addEventListener('click', (e) => {
    e.preventDefault();
    const inputType = document.getElementById('passwordUser');
    if (inputType.type === 'password') {
      inputType.type = 'text';
      togglePassword.src = 'images/eye-slash-regular.jpeg';
    } else {
      inputType.type = 'password';
      togglePassword.src = 'images/eye-regular.jpeg';
    }
  });
  // Tercer hijo del form enviar
  buttonSignup.setAttribute('type', 'submit');
  buttonSignup.setAttribute('class', 'signup-button');
  buttonSignup.id = 'submitBTN';
  buttonSignup.textContent = 'Regístrate';
  signupForm.appendChild(buttonSignup);
  // Esta es la parte del firebase
  buttonSignup.addEventListener('click', (e) => {
    e.preventDefault();

    const email = signupForm.querySelector('#emailUser').value;
    const password = signupForm.querySelector('#passwordUser').value;

    signUp(email, password)
      .then(() => {
        alert('usuario registrado');
        onNavigate('/feed');
      })
      .catch((error) => {
        const errorMessage = error.message;
        alert(errorMessage);
      });
  });
  sectionChild.appendChild(spacebetween());
  sectionChild.appendChild(ButtonGoogle());
  // Este es el espacio entre botones
  sectionChild.appendChild(spaceButtons);
  // Este es el boton de redireccion a login
  loginButton.setAttribute('class', 'login-in-register');
  loginButton.setAttribute('type', 'button');
  loginButton.textContent = 'Inicia Sesion Aquí';
  sectionChild.appendChild(loginButton);
  loginButton.addEventListener('click', (e) => {
    e.preventDefault();
    onNavigate('/');
  });
  return signupView;
};
