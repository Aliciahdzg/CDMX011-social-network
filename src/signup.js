const signup = `
<section class="grid-father">
    <section class="container ">
            <form id="signup ">
                <label>Correo</label>
                    <input type="email " id="emailUser " class="form-elements " placeholder="Correo " name="Email " required />
                <label>Contraseña</label>
                <input type="password " id="passwordUser " class="form-elements " placeholder="Contraseña " name="password-user " required>
                <button type="submit " value="enviar " id="submitBTN " class="signup-button ">Regístrate</button>
            </form>
            <p class="hr-test "><span>ó</span></p>
            <button type="button " id="registerGoogle " class="signup-google button-google "> 
             <img src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg "><div>Accede con Google</div></button>
            <br>
            <button type="button " id="loginInRegister " class="login-in-register ">Inicia Sesion Aqui</p>
    </section>
</section>
`;