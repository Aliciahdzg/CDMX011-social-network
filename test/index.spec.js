/**
 * @jest-environment jsdom
 */
import './globals/firebaseTest.js';
import { Signup } from '../src/components/signup.js';

window.alert = jest.fn();

describe('Signup', () => {
  const component = Signup();
  it('should render signup view', () => {
    expect(component.outerHTML).toMatchSnapshot();
  });
  it('should register the user when the submit button is clicked', async () => {
    window.alert.mockClear();

    const mockSignUp = jest.fn();
    mockSignUp.mockImplementation(() => Promise.resolve());

    firebase.auth = jest.fn().mockImplementation(() => ({
      createUserWithEmailAndPassword: mockSignUp,
    }));

    const email = 'user@example.com';
    const password = '123456';

    component.querySelector('#emailUser').value = email;
    component.querySelector('#passwordUser').value = password;
    const submitBTN = component.querySelector('#submitBTN');
    submitBTN.click();
    await Promise.resolve();
    expect(mockSignUp).toHaveBeenCalled();
  });
  it('should show password and replace button img by an slash-eye', () => {
    // const displayPassword = component.querySelector('#displayPassword');
    const togglePassword = component.querySelector('#togglePassword');
    togglePassword.click();
    expect(togglePassword).toMatchSnapshot();
  });
});
