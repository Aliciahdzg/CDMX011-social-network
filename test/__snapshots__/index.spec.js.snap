// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should render signup view 1`] = `"<section class=\\"grid-father\\"><section class=\\"container\\"><h1 class=\\"logo\\" id=\\"logo-title\\">Plantagram</h1><img src=\\"images/logo.png\\" alt=\\"plantita\\" class=\\"image-size\\" id=\\"logo-img\\"><img src=\\"images/girls.jpg\\" alt=\\"chicas\\" class=\\"image-size\\" id=\\"logo-img-land\\"><p class=\\"welcome-text\\">Bienvenid@ a la plataforma destinada a los amantes de plantas</p></section><section class=\\"container\\"><form id=\\"signup\\"><input type=\\"email\\" class=\\"form-elements\\" placeholder=\\"Correo\\" name=\\"email\\" id=\\"emailUser\\"><div class=\\"form-elements\\" id=\\"displayPassword\\"><input type=\\"password\\" placeholder=\\"Contraseña\\" class=\\"input-password\\" name=\\"password-user\\" id=\\"passwordUser\\" autocomplete=\\"on\\"><img id=\\"togglePassword\\" class=\\"eye-img\\" src=\\"images/eye-regular.jpeg\\"></div><button type=\\"submit\\" class=\\"signup-button\\" id=\\"submitBTN\\">Regístrate</button></form><p class=\\"hr-test\\"><span>ó</span></p><button type=\\"button\\" class=\\"signup-google button-google\\" id=\\"registerGoogle\\"><img src=\\"https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg \\"><div class=\\"access-google\\">Accede con Google</div></button><br><button class=\\"login-in-register\\" type=\\"button\\">Inicia Sesion Aquí</button></section></section>"`;

exports[`Signup should show password and replace button img by an slash-eye 1`] = `
<img
  class="eye-img"
  id="togglePassword"
  src="images/eye-regular.jpeg"
/>
`;
